datadog_checks_dev/datadog_checks/dev/plugin/pytest.py

```diff
@@ -85,3 +104,76 @@ def dd_environment_runner(request):
     else:  # no cov
         # Exit testing and pass data back up to command
         pytest.exit(message)
+
+
+@pytest.fixture
+def dd_agent_check(request, aggregator):
+    if not e2e_testing():
+        pytest.skip('Not running E2E tests')
+
+    # Lazily import to reduce plugin load times for everyone
+    from datadog_checks.dev import TempDir, run_command
+
+    def run_check(config, **kwargs):
+        root = os.path.dirname(request.module.__file__)
+        while True:
+            if os.path.isfile(os.path.join(root, 'setup.py')):
+                check = os.path.basename(root)
+                break
+
+            new_root = os.path.dirname(root)
+            if new_root == root:
+                raise OSError('No Datadog Agent check found')
+
+            root = new_root
+
+        config = format_config(config)
+        env = os.environ['TOX_ENV_NAME']
+        python_path = os.environ[E2E_PARENT_PYTHON]
+        config_file = os.path.join(temp_dir, '{}-{}-{}.json'.format(check, env, urlsafe_b64encode(os.urandom(6))))
+
+        with open(config_file, 'wb') as f:
+            output = json.dumps(config).encode('utf-8')
+            f.write(output)
+
+        check_command = [
+            python_path,
+            '-m',
+            'datadog_checks.dev',
+            'env',
+            'check',
+            check,
+            env,
+            '--config',
+            config_file,
+            '--json',
+        ]
```

Comment:
Do we really require that much stuff when calling ddev? Aren't those imports available in the env anyway? It would make code much simpler if we could just call the function directly.

Reply:
We separate the deps

https://github.com/DataDog/integrations-core/blob/8ead7b17ed3045e6feeb38f484f51a2ecdf72bfa/datadog_checks_dev/setup.py#L24-L34
https://github.com/DataDog/integrations-core/blob/8ead7b17ed3045e6feeb38f484f51a2ecdf72bfa/datadog_checks_dev/setup.py#L64-L82

Reply:
Couldn't the problem be worked around with late imports?

Reply:
True but that would have to happen everywhere and would look odd certainly

Reply:
I meant just for that particular call.

Reply:
Oh then no due to how command discovery works